### DISCUSSION TITLE
Blank PDF generated - local machine & self-signed certificate, Docker compose, ASP.NET Core, nginx [migration from 6 to 7]

### DISCUSSION THREAD

**Discussion by ivan-84** [gotenberg/gotenberg#513] [Q&A] [answered]
Hi Julien and all,

First of all let me say that you've done (I mean Julien and all the conributors) a truly amazing job with this project - thank you very much for your efforts!
Now, back to the subject. I'll simply describe the architecure so that it's clear what I was trying to solve: application consists of serveral Docker containers and is built with Docker compose file - I have Rest API container (not that it matters, but it's a .NET 6 Web API project), React for front-end (built with CRA), MS SQL Server, Gotenberg and nginx as a reverse proxy to put it all together. A couple of days ago, I've started  upgrading solution to .NET 6 (from .NET Core 3.1) and I've also upgraded Gotenberg Sharp Client to the latest version which is designed to work with Gotenberg 7+ - so I've upgraded Gotenberg too. This is where my problems have started :) With version 6 I've had in Docker compose file in "environment" directive defined "GOOGLE_CHROME_IGNORE_CERTIFICATE_ERRORS: "1"" (which I use of course only on my local machine) and everything was working as expected - I was sending URL of the nginx to Gotenberg for conversion to PDF and I was getting a generated PDF. But after the upgrade to the version 7, suddenly I've started to get blank page instead of generated file, which reminded me the same issue with version 6 before I've put that environmental variable into Docker compose. So I have been fighting with self-signed certificate on my machine (Win 10) because I thought that it's not trusted or expired - but the problem of course was elsewhere as you might've already guessed, Julien :) Thing is that I was referring to the official documentation on https://gotenberg.dev/ and unfortunately there was no information how to make Gotenberg (or rather Chromium) to ignore certificate errors, nor I've found anything on GitHub / Stackoverflow or simple googling. I was desperate as I've already burnt more than one day trying to solve this issue. Today as refreshed and rested man :) I have decided to dig into the source files here on GitHub and bingo - after a couple of minutes I've found in Makefile, that Gotenberg no longer uses Chrome, but Chromium and that the environmental variable should be "CHROMIUM_IGNORE_CERTIFICATE_ERRORS: true"...but adding that to the Docker compose didn't work :) So I've kept digging and came to the conclusion that I actually need that to be passed as a parameter to the Gotenberg. After modifying my Docker file like this:

`FROM thecodingmachine/gotenberg:7
EXPOSE 3000
CMD ["gotenberg", "--chromium-ignore-certificate-errors=true"]`

all started to work again! 

**That said, PLEASE, Julien, I beg of you, add this "gotcha" to the documentation on the site - I'm pretty sure it can help a lot of developers with similar project architecture!**

**Comment by gulien** [comment, marked as answer]
Hello @ivan-84,

Sorry for the delay. It's actually already in the documentation: https://gotenberg.dev/docs/modules/chromium#properties 😄

**Reply by ivan-84** [reply]
Hi @gulien ,

Thank you for reply. Indeed, the documentation is mentioning that flag - somehow I've managed to overlook it. :) But still maybe it would make sense to create a small page with info related to migration from 6 to 7? In version 6 it was clear that in order to ignore certificate errors an environment variable should be defined, but in 7 IMO it's not that obvious that one should actually run Gotenberg itself with some flag. Or we can leave it with hope that uncle Google will index this discussion :)

**Reply by gulien** [reply]
Good point! Alas, I do not have much time available currently, having a young child to manage in my free time 😄 

That being said, I thought it was pretty obvious about the flag, as I speak about them early in the documentation. For instance, Docker Compose: https://gotenberg.dev/docs/get-started/docker-compose#modules-properties

**Reply by ivan-84** [reply]
I feel your pain, got two kids :) Anyways, it was just a suggestion - somehow I've managed to overlook that and I wasn't even sure what's going on because I got no errors in Gotenberg container's log (even though I've set LOG_LEVEL to "DEBUG") and at the same time I kept getting blank PDF as a result. In any case I'd like to express my cencere gratitude and respect for this wonderful project - keep it up! 🥇

**Reply by gulien** [reply]
Thanks for the kind words!
